astropy/table/table.py

```diff
@@ -618,6 +623,39 @@ def __init__(self, data=None, masked=False, names=None, dtype=None,
         if self.masked not in (None, True, False):
             raise TypeError("masked property must be None, True or False")
 
+        self._set_column_attribute('unit', units)
+        self._set_column_attribute('description', descriptions)
+
+    def _set_column_attribute(self, attr, values):
+        """Set ``attr`` for columns to ``values``, which can be either a dict (keyed by column
+        name) or a dict of name: value pairs.  This is used for handling the ``units`` and
+        ``descriptions`` kwargs to ``__init__``.
+        """
+        if not values:
+            return
+
+        if isinstance(values, Row):
```

Comment:
For future, perhaps we should have `Rows.items()` so there isn't a need to special-case it.